libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1460,6 +1487,35 @@ void AP_GPS::send_mavlink_gps_rtk(mavlink_channel_t chan, uint8_t inst)
 }
 #endif
 
+void AP_GPS::send_mavlink_gnss_integrity(mavlink_channel_t chan, uint8_t instance) {
+    uint8_t id = instance; 
+    uint32_t system_error  = get_system_errors(instance);
+    uint8_t authentication = get_authentication_state(instance);
+    uint8_t jamming        = get_jamming_state(instance);
+    uint8_t spoofing       = get_spoofing_state(instance);
+    uint8_t raim_state     = UINT8_MAX; //not implemented yet 
+    uint16_t raim_hfom     = UINT16_MAX; //not implemented yet
+    uint16_t raim_vfom     = UINT16_MAX; //not implemented yet
+    uint8_t corrections_quality     = UINT8_MAX; //not implemented yet
+    uint8_t systems_status_summary  = UINT8_MAX; //not implemented yet
+    uint8_t gnss_signal_quality     = UINT8_MAX; //not implemented yet
+    uint8_t post_processing_quality = UINT8_MAX; //not implemented yet
```

Comment:
Don't bother making these locals, just throw them straight into the packet; add comments to indicate which field is being passed where things are unclear.